lib/internal/url.js

```diff
@@ -1400,6 +1400,33 @@ function getPathFromURL(path) {
   return isWindows ? getPathFromURLWin32(path) : getPathFromURLPosix(path);
 }
 
+function NativeURL(ctx) {
+  this[context] = ctx;
+}
+util.inherits(NativeURL, URL);
```

Comment:
I think this would lead to observable differences between URLs constructed with `URL` vs `NativeURL`. How about a plain `NativeURL.prototype = URL.prototype`?